profiles/common/modules/features/nexteuropa_geofield/nexteuropa_geofield.module

```diff
@@ -322,6 +356,85 @@ function _nexteuropa_geofield_get_geojson($node) {
 }
 
 /**
+ * Ajax callback to fetch the available coordinates for specified address.
+ *
+ * @param array $form
+ *   Drupal form array.
+ * @param array $form_state
+ *   Drupal form state array.
+ *
+ * @return string
+ *   Return the Ajax response to be rendered in the form.
+ */
+function _nexteuropa_geofield_geocode($form, $form_state) {
+
+  $error = NULL;
+  $response = '';
+
+  $geo_field_name = $form_state['triggering_element']['#attributes']['search-field'];
+  $address = $form_state['values'][$geo_field_name][LANGUAGE_NONE][0]['geojson_search'];
+
+  if (empty($address)) {
+    $error = t('No address specified for search.');
+  }
+  else {
+    /*
+     * @todo move to configuration.
+     */
+    $url = 'http://europa.eu/webtools/rest/geocoding/index.php?address=' . urlencode($address) . '&mode=5';
+    // Contact the WebTools geocoding service.
+    $geocode = drupal_http_request($url);
+
+    if (!empty($geocode->error)) {
+      $error = $geocode->error;
+    }
+    else {
+      $geocode_decoded = json_decode($geocode->data);
+
+      if ($geocode_decoded->geocodingRequestsCollection[0]->foundCount < 1) {
+        $error = t('No address found.');
+      }
+      else {
+        $added_items = array();
+        $items = array();
+        foreach ($geocode_decoded->geocodingRequestsCollection[0]->result->features as $key => $address) {
+          $geometry = $address->geometry->coordinates;
+          $geometry_string = $geo_field_name[0] . ',' . $geometry[1];
+          // Don't add address if already added. Prevent duplicates.
+          if (array_search($geometry_string, $added_items) === FALSE) {
```

Comment:
Instead of doing that I would compute a `sha1()` of `$geometry_string` and use it as key in the array so we prevent duplicate without adding complexity.

```php
$key = sha1($geometry_string);
$items[$key] = array(...);
```